controllers/workshop_controller.go

```diff
@@ -246,10 +223,21 @@ func (r *WorkshopReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop) (ctrl.Result, error) {
+func (r *WorkshopReconciler) handleDelete(ctx context.Context, req ctrl.Request, workshop *workshopv1.Workshop, userID int, appsHostnameSuffix string, openshiftConsoleURL string) (ctrl.Result, error) {
 	log := r.Log.WithValues("workshop", req.NamespacedName)
 	log.Info("Deleting workshop" + workshop.ObjectMeta.Name)
 
+	if result, err := r.deleteVaultServer(workshop); util.IsRequeued(result, err) {
+		return result, err
+	}
+	if result, err := r.deleteVaultAgentInjector(workshop); util.IsRequeued(result, err) {
+		return result, err
+	}
+
+	if result, err := r.deleteVaultNamespace(workshop); util.IsRequeued(result, err) {
+		return result, err
+	}
+
```

Comment:
@MaazMS this is absolutely wrong; you should add ONE method in vault which handles all these three methods

https://github.com/stakater/workshop-operator/pull/31/files#diff-239a055a1843001a63725aaf6c3ef73cc8d09de34f654bd8cd62b2d3914e2ee5R230-R240

and in the workshop controller you should just call that ONE method

Reply:
@rasheedamir  i will create single method, in workshop controller ,